pkg/controller/storagecluster/storagecluster.go

```diff
@@ -773,10 +759,34 @@ func (c *Controller) manage(
 			zoneMap[zone] = instancesCount + 1
 		}
 	}
+	return nodeList, zoneMap, nil
+}
+
+func (c *Controller) manage(
+	cluster *corev1.StorageCluster,
+	hash string,
+) error {
+	// Run the pre install hook for the driver to ensure we are ready to create storage pods
+	if err := c.Driver.PreInstall(cluster); err != nil {
+		return fmt.Errorf("pre-install hook failed: %v", err)
+	}
+
+	// Find out the pods which are created for the nodes by StorageCluster
+	nodeToStoragePods, err := c.getNodeToStoragePods(cluster)
+	if err != nil {
+		return fmt.Errorf("couldn't get node to storage cluster pods mapping for storage cluster %v: %v",
+			cluster.Name, err)
+	}
+	var nodesNeedingStoragePods, podsToDelete []string
+
+	nodeList, zoneMap, err := c.getZoneMap(cluster)
+	if err != nil {
+		return err
+	}
```

Comment:
nit: We can just call the UpdateDriver from the `SetDefaults` function. We don't need to do it here. 
Here, we just need the list of nodes. I would suggest just getting the node list and passing it to getZoneMap method. We don't even need to store the cloudProviderName that way.